src/Microsoft.Extensions.Options/OptionsSnapshot.cs

```diff
@@ -1,34 +1,21 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
+
 namespace Microsoft.Extensions.Options
 {
     /// <summary>
     /// Implementation of IOptionsSnapshot.
     /// </summary>
     /// <typeparam name="TOptions"></typeparam>
-    public class OptionsSnapshot<TOptions> : IOptionsSnapshot<TOptions> where TOptions : class, new()
+    public class OptionsSnapshot<TOptions> : OptionsManager<TOptions>, IOptionsSnapshot<TOptions> where TOptions : class, new()
     {
-        private TOptions _options;
-
         /// <summary>
         /// Initializes a new instance.
         /// </summary>
-        /// <param name="monitor">The monitor to fetch the options value from.</param>
-        public OptionsSnapshot(IOptionsMonitor<TOptions> monitor)
-        {
-            _options = monitor.CurrentValue;
-        }
-
-        /// <summary>
-        /// The configured options instance.
-        /// </summary>
-        public virtual TOptions Value
-        {
-            get
-            {
-                return _options;
-            }
-        }
+        /// <param name="setups">The configuration actions to run.</param>
+        public OptionsSnapshot(IEnumerable<IConfigureOptions<TOptions>> setups) : base(setups)
+        { }
```

Comment:
Slight behavior difference with this change, previously we would eager cache .Value in the ctor, now we are lazy and don't cache until the first time .Value is used...  We should be intentional with how this works I guess...